src/epics/router.ts

```diff
@@ -188,6 +229,53 @@ const handleRouter: Epic = (action$, store) => action$.ofType(types.handleRouter
     })
 })
 
+function parseFiltersFromString(defaultFilters: string) {
+    const result: BcFilter[] = []
+    const urlParams = qs.parse(defaultFilters)
+    Object.keys(urlParams).forEach((param) => {
+        const [fieldName, type] = param.split('.')
+        if (fieldName && type && urlParams[param]) {
+            let value = urlParams[param]
+            if (isManyValueFilterType(type)) {
+                try {
+                    value = JSON.parse(value)
+                } catch (e) {
+                    console.warn(e)
+                }
+                value = Array.isArray(value) ? value : []
+            }
+            result.push({
+                fieldName,
+                type: type as FilterType,
+                value
+            })
+        }
+    })
+    return result.length ? result : null
+}
+
+function isManyValueFilterType(type: string) {
+    return (
+        type === FilterType.containsOneOf ||
+        type === FilterType.equalsOneOf
+    )
+}
+
+function locationChanged(pathname: string, currentScreenName: string, currentViewName: string)  {
```

Comment:
[There is a function](https://github.com/tesler-platform/tesler-ui/blob/a0ac06237d57cfa359fdb26ffb7954b8b56b62e9/src/Provider.tsx#L123) that is used for splitting into the `Route` object. This function could be overridden from client side as long as contract have been followed, i.e. it does not necessary will work with this specific splitting algorithm.

`locationChanged` should be called with two `Route` arguments instead

Reply:
Fixed, function locationChanged removed